Source/Pawnmorphs/Esoteria/Hediffs/Composable/TFTypes.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using Verse;
+
+namespace Pawnmorph.Hediffs.Composable
+{
+    /// <summary>
+    /// A class that determines what kind(s) of animals a pawn can be transformed into
+    /// </summary>
+    public abstract class TFTypes
+    {
+        /// <summary>
+        /// Gets a pawn kind to transform the pawn into
+        /// </summary>
+        /// <returns>The mutations.</returns>
+        /// <param name="hediff">Hediff.</param>
+        public abstract PawnKindDef GetTF(Hediff_MutagenicBase hediff);
+
+        /// <summary>
+        /// A debug string printed out when inspecting the hediffs
+        /// </summary>
+        /// <param name="hediff">The parent hediff.</param>
+        /// <returns>The string.</returns>
+        public virtual string DebugString(Hediff_MutagenicBase hediff) => "";
+    }
+
+    /// <summary>
+    /// A simple TFTypes that allows a transformation into ALL THE ANIMALS _O/
+    /// Good for chaotic mutations.
+    /// </summary>
+    public class TFTypes_All : TFTypes
+    {
+        /// <summary>
+        /// Gets a pawn kind to transform the pawn into
+        /// </summary>
+        /// <returns>The mutations.</returns>
+        /// <param name="hediff">Hediff.</param>
+        public override PawnKindDef GetTF(Hediff_MutagenicBase hediff)
+        {
+            return DefDatabase<PawnKindDef>.AllDefs
```

Comment:
I added a utility function to FormerHumanUtilities and ChaormorphUtilities that contain a cached list of all possible random animals and chaomorphs a pawn can tf into respectively. Better to use that, as some ‘animals’ have been patched to never be former humans, specifically from mods like android tiers